File: src/org/symbolmath/ast/leaf/IntegerElement.java
package org.symbolmath.ast.leaf;

import org.symbolmath.scanner.Token;

public class IntegerElement extends LeafElement {
  private Integer value;

  public IntegerElement(Token token) {
    this.value = (Integer) token.getValue();
  }

  @Override
  public String toString() {
    return value.toString();
  }

  public int getValue() {
    return value;
  }
}

File: src/main/java/org/symbolmath/ast/leaf/IntegerElement.kt
package org.symbolmath.ast.leaf

class IntegerElement(val value: Int) : LeafElement() {
    override fun toString(): String {
        return value.toString()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as IntegerElement

        if (value != other.value) return false

        return true
    }

    override fun hashCode(): Int {
        return value
    }


}